cli/src/main/kotlin/OrtMain.kt

```diff
@@ -80,7 +81,11 @@ data class GlobalOptions(
  */
 fun main(args: Array<String>) {
     Os.fixupUserHomeProperty()
-    OrtMain().main(args)
+    try {
+        OrtMain().main(args)
+    } finally {
+        println("ORT command execution terminated at ${LocalDateTime.now()}.")
```

Comment:
This is know printed only right before `exitProcess(0)`.
It is not printed if ORT exits in a different place. There must be some `exitProcess(1)` or `exitProcess(2)` calls elsewhere. So, could there be a better place to move the printing to?

Reply:
> There must be some `exitProcess(1)` or `exitProcess(2)` calls elsewhere.

No, there aren't. *clikt* never calls `exit(Process)` itself but throws `ProgramResult` exceptions to terminate and return to the caller (it's IMO bad that *clikt* uses exception for programm flow here, but that's a different story).

Reply:
Do you have any proposals?

I currently don't have the time to invest deeper here. I would also be fine to redraw this PR if you think it is not useful in its current state

Reply:
> No, there aren't. _clikt_ never calls `exit(Process)` itself but throws `ProgramResult` exceptions to terminate and return to the caller

That's not exactly correct, see https://github.com/ajalt/clikt/blob/c9e7757e846f1f75fbe15488863c20b481116f1e/clikt/src/commonMain/kotlin/com/github/ajalt/clikt/core/CliktCommand.kt#L414-L437.

So, in case a `ProgramResult` is thrown, the added print statement won't be executed.

Reply:
From the above, maybe the whole code inside `run()` should be moved into a ´try` and then `finally` the statement should be printed inside `run()`?!

Reply:
> That's not exactly correct

You're right.

> Do you have any proposals?

The only thing that comes to my mind is to call `CliktCommand.parse()` ourtselves directly but that's also not feasible

> I would also be fine to redraw this PR if you think it is not useful in its current state

I indeed fear the PR does not do anything meaningful (on application error) in its current state.

Reply:
> From the above, maybe the whole code inside `run()` should be moved into a `try` and then `finally` the statement should be printed inside `run()`?!

IIUC you'd have to to this for all subcommands then, not just `OrtMain`, which IMO is not nice...

Reply:
It helped me at least to debug cases when Kubernetes has killed a pod. But as said, I am fine to close this; it won't be possible to cover all scenarios.

Reply:
> It helped me at least to debug cases when Kubernetes has killed a pod.

Ah, then I might have misunderstood: I thought the main purpose was to add the output when ORT exists with code != 0. If that's not the case, and this output had helped you in the past, I'm still fine with merging this.

Reply:
> maybe the whole code inside `run()` should be moved into a `try` and then `finally` the statement should be printed inside `run()`?!

Wouldn't that approach work? (e.g. only in the run function of the main command)

Reply:
> Wouldn't that approach work? (e.g. only in the run function of the main command)

I guess not, because a subcommand's `run` is not called from the main command's `run`, so if an exception occurs in a subcommand's `run` it's not propagated back to the main command's `run`.

Reply:
I see, thanks for explaining.

Reply:
I just found https://docs.oracle.com/javase/8/docs/api/java/lang/Runtime.html#addShutdownHook-java.lang.Thread-, not sure if it's a good idea to use it for that purpose though.